packages/models-library/src/models_library/api_schemas_dynamic_sidecar/telemetry.py

```diff
@@ -28,31 +36,71 @@ def percent(self) -> float:
         ...
 
 
+def _get_percent(used: float, total: float) -> float:
+    return round(used * 100 / (total + _EPSILON), 2)
+
+
 class DiskUsage(BaseModel):
     used: ByteSize = Field(description="used space")
     free: ByteSize = Field(description="remaining space")
 
     total: ByteSize = Field(description="total space = free + used")
-    used_percent: float = Field(
+    used_percent: NonNegativeFloat = Field(
         gte=0.00,
         lte=100.00,
         description="Percent of used space relative to the total space",
     )
 
+    @validator("free")
     @classmethod
-    def from_ps_util_disk_usage(
-        cls, ps_util_disk_usage: SDiskUsageProtocol
+    def _free_positive(cls, v: float) -> float:
```

Comment:
ByteSize already does not allow for negative values when validated. so why is this validator needed?

Reply:
Only in pydantic2, it still accepts negative values in pydantic1. checked it and tests do not pass.